### PR TITLE
Add debugger smoke tests project

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Add smoke tests for Live Debugger product

# Motivation
Open sourcing Live Debugger product

### CODE REVIEW

**Review by bantonsson — COMMENTED** [review]
I can't say much about the actual tests (so I have to trust you), but I have some minor comments/questions.

File: dd-smoke-tests/debugger-integration-tests/latest-jdk-app/src/main/java/App.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        waitForSnapshotUpload(1);
+        System.out.println("Exiting...");
+    }
+
+  private static void waitForInstrumentation() throws IOException {
+    AtomicBoolean generatingByteCode = new AtomicBoolean();
+    waitForSpecificLogLine(
+      Paths.get(LOG_FILENAME),
+      line -> {
+        // when instrumentatiob is done by main thread, we are good to go
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-smoke-tests/debugger-integration-tests/latest-jdk-app/src/main/java/App.java]: done

File: dd-smoke-tests/debugger-integration-tests/debugger-integration-tests.gradle [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
+
+tasks.register('buildLatestJdkApp', GradleBuild) {
+  buildFile = 'latest-jdk-app/latest-jdk-app.gradle'
+  tasks = ['shadowJar']
+}
+
+def latestJdk = '16'
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-smoke-tests/debugger-integration-tests/debugger-integration-tests.gradle]: yes! done.

File: dd-smoke-tests/debugger-integration-tests/debugger-integration-tests.gradle [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+tasks.withType(Test).configureEach {
+  dependsOn shadowJar
+
+  def isLatestJdk = it.name.contains(latestJdk)
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-smoke-tests/debugger-integration-tests/debugger-integration-tests.gradle]: ah yes Thanks for the tip!

File: dd-smoke-tests/debugger-integration-tests/latest-jdk-app/.gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Ignore Gradle build output directory
+build
+.gradle
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-smoke-tests/debugger-integration-tests/latest-jdk-app/.gitignore]: done

**Pull request merged**